astroid/decorators.py

```diff
@@ -99,9 +99,8 @@ def wrapped(node, context=None, _func=func, **kwargs):
             # Explicit StopIteration to return error information, see
             # comment in raise_if_nothing_inferred.
             if error.args:
-                raise StopIteration(error.args[0])
-            else:
-                raise StopIteration
+                return error.args[0]
+            return
```

Comment:
The tests seem to be passing on 3.7 with these changes. The only thing that is failing is some linting errors:
`return None` here

Reply:
Yup, as mentioned, I think we run on an "old" version of 3.7 on Travis, probably alpha 1 or so. I have it locally and the beta 1 seems to have this change, so this could explain why the tests are passing on Travis. From a short look, it doesn't seem that there is an update on Travis.